pkgs/development/libraries/hyperscan/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv
+, fetchurl
+, pkgconfig
+, boost
+, cmake
+, python
+, ragel
+}:
+
+let
+
+in
+stdenv.mkDerivation rec {
+  name = "hyperscan-${version}";
+  version = "4.6.0";
+
+  src = fetchurl {
+    url = "https://github.com/intel/hyperscan/archive/v${version}.tar.gz";
+    sha256 = "05d8zpjcm13lldkzcxafkabllzgv10nrhgfazsrag9l2bqpgryqd";
+  };
+
+  nativeBuildInputs = [
+    boost
+    cmake
+    pkgconfig
+    python
+    ragel
+  ];
+
+  enableParallelBuilding = true;
+
+  cmakeFlags = [
+    "-DBUILD_AVX512=ON"
+    "-DBUILD_SHARED_LIBS=ON"
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://01.org/hyperscan;
+    description = "A high-performance multiple regex matching library";
+    license = [ licenses.bsd3 licenses.bsd2 licenses.boost ];
```

Comment:
This could be simplified as `license = with licenses; [ bsd3 bsd2 boost ];`

Reply:
`with licenses; [ ... boost ]` does not evaluate, FYI, because `boost` is also included in the function arguments.

Reply:
`with` clause will take precedence and shadow function arguments in that case.

Reply:
Try it for yourself.

https://gist.github.com/dhess/ec32c05364c73a69b91813d0d4e7e71d

```
$ nix-build pkgs/top-level/release.nix -A hyperscan
trace: WARNING: `addPassthru` is deprecated, replace with `extendDerivation true`
trace: lib.zip is deprecated, use lib.zipAttrsWith instead
trace: `mkStrict' is obsolete; use `mkOverride 0' instead.
trace: `types.list` is deprecated; use `types.listOf` instead
trace: types.optionSet is deprecated; use types.submodule instead
error: Package ‘python2.7-dosage-2016.03.17’ in /home/dhess/git/nixpkgs/pkgs/top-level/python-packages.nix:1555 is marked as broken, refusing to evaluate.

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.

(use ‘--show-trace’ to show detailed location information)
```

Reply:
You are right; interesting.

This prints `2`:

```
let
  a = { foo = 1; };
  b = { foo = 2; };
in

with a;
with b;

builtins.trace foo foo
```

And this prints `1`:

```
{ foo ? 1 }:

let
  b = { foo = 2; };
in

with b;

builtins.trace foo foo
```

I'm not sure if that's intended; that doesn't seem to follow proper scoping rules. If there is no ticket against NixOS/nix on this, perhaps one should be opened.

Reply:
Found it, see https://github.com/NixOS/nix/issues/490.